packages/hooks/src/useConfig.ts

```diff
@@ -1,8 +1,28 @@
-import { useContext } from 'react'
+import { useContext, useEffect, useState } from 'react'
 import { LiteflowContext } from './context'
 import { Config } from './graphql'
 
-export default function useConfig(): Promise<Config> {
-  const { config } = useContext(LiteflowContext)
-  return config
+// This type can be reused for any other read-only hook
+type Result<T = any> = {
+  data?: T
+  error?: Error
+}
+
+export default function useConfig(): Result<Config> {
+  const { sdk } = useContext(LiteflowContext)
+  const [config, setConfig] = useState<Config>()
+  const [error, setError] = useState<Error>()
+
+  useEffect(() => {
+    sdk
+      .GetConfig()
+      .then(({ config }) => config)
+      .catch(setError)
+    return () => setConfig(undefined)
```

Comment:
It's good practice to reset the states in useEffects. In that case, if the sdk changes this is because the endpoint changes so the config should be refetched. If we don't reset it we will have during the fetching time invalid configuration for this new endpoint